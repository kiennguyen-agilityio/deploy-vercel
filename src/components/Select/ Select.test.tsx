import { render, fireEvent, screen } from '@testing-library/react';
import ColorSelector from './Select';

test('renders without crashing', () => {
  const { container } = render(<ColorSelector onColorChange={() => {}} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('calls onColorChange when color is selected', () => {
  const mockOnColorChange = jest.fn();

  render(
    <select onChange={mockOnColorChange}>
      <option value="Black">Black</option>
      <option value="Red">Red</option>
      <option value="Green">Green</option>
      <option value="Yellow">Yellow</option>
    </select>
  );

  const selectElement = screen.getByRole('combobox') as HTMLSelectElement;

  selectElement.value = 'Red';
  fireEvent.change(selectElement);

  expect(mockOnColorChange).toHaveBeenCalledTimes(1);
});
