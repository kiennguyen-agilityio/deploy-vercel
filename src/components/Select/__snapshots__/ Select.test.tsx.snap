// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="product-color"
>
  <p>
    Color
  </p>
  <div
    class="select-wrapper"
  >
    <select>
      <option
        value="Black"
      >
        Black
      </option>
      <option
        value="Red"
      >
        Red
      </option>
      <option
        value="Green"
      >
        Green
      </option>
      <option
        value="Yellow"
      >
        Yellow
      </option>
    </select>
  </div>
</div>
`;
